packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -127,16 +128,12 @@ export async function runHttpQuery(
     // debug. Therefore, we need to do some unnatural things, such
     // as use NODE_ENV to determine the debug settings
     return throwHttpGraphQLError(500, [e], {
-      debug: debugFromNodeEnv(process.env.NODE_ENV),
+      debug: debugFromNodeEnv(),
     });
```

Comment:
Based on the comment above this, I don't think we can actually use what is in the configuration.

Reply:
I tried something, but I don't think it's technically correct. With how its setup, it's technically possible to provide `nodeEnv` asynchronously.

For this one, I think we may have to use the solution from the first commit.